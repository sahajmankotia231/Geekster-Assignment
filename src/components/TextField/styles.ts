import styled from "styled-components";

import { BORDER_GRAY, BLUE } from "constants/colors";
import { INPUT_RADIUS } from "constants/styles";

export const InputContainer = styled.div`
  width: 100%;
  position: relative;
`;

export const LeftIcon = styled.i`
  position: absolute;
  top: 12px;
  left: 12px;
`;

export const RightIcon = styled.i`
  position: absolute;
  top: 12px;
  right: 12px;
`;

export const Input = styled.input`
  width: 100%;
  height: 40px;
  border: 1px solid ${BORDER_GRAY};
  border-radius: ${INPUT_RADIUS}px;
  display: flex;
  font-size: 15px;
  line-height: 24px;
  padding-left: ${({ leftIcon }) => (leftIcon ? "36px" : "8px")};
  padding-right: ${({ rightIcon }) => (rightIcon ? "36px" : "8px")};

  &:focus {
    border: 1px solid ${BLUE};
    box-shadow: 0 0 1px ${BLUE};
    outline: none;
    transition: all ease 0.2s;
  }
`;
