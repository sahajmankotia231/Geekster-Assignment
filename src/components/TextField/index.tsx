import React, { ReactElement } from "react";

import { Input, InputContainer, LeftIcon, RightIcon } from "./styles";

interface TextFieldProps {
  placeholder?: string;
  leftIcon?: ReactElement;
  rightIcon?: ReactElement;
  type?: string;
  min?: number;
  max?: number;
  readonly?: boolean;
  onClick?: () => void;
  onChange: () => void;
  value?: string;
}

const TextField = ({
  placeholder,
  leftIcon,
  rightIcon,
  type = "text",
  min,
  max,
  readonly,
  onClick,
  onChange,
  value = "",
}: TextFieldProps): ReactElement => (
  <InputContainer>
    {leftIcon && <LeftIcon onClick={onClick}>{leftIcon}</LeftIcon>}
    <Input
      type={type}
      placeholder={placeholder}
      leftIcon={leftIcon}
      rightIcon={rightIcon}
      min={min}
      max={max}
      readonly={readonly}
      onClick={onClick}
      onChange={onChange}
      value={value}
    />
    {rightIcon && <RightIcon>{rightIcon}</RightIcon>}
  </InputContainer>
);

export default TextField;
