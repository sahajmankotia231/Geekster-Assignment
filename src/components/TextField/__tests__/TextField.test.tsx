import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";

import TextField from "..";

describe("<TextField />", () => {
  it("should render properly", () => {
    const { container } = render(<TextField onChange={() => {}} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should be able to render a number", () => {
    const placeholder = "Guests";
    render(
      <TextField onChange={() => {}} type="number" placeholder={placeholder} />
    );
    expect(screen.getByPlaceholderText(placeholder).getAttribute("type")).toBe(
      "number"
    );
  });

  it("should trigger onClick", () => {
    const mockedOnClick = jest.fn();
    const placeholder = "Guests";

    render(
      <TextField
        onChange={() => {}}
        onClick={mockedOnClick}
        placeholder={placeholder}
      />
    );

    fireEvent.click(screen.getByPlaceholderText(placeholder));

    expect(mockedOnClick).toHaveBeenCalled();
  });
});
