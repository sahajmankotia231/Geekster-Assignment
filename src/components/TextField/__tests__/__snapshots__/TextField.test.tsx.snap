// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> should render properly 1`] = `
<div
  class="sc-bdvvaa jSVYap"
>
  <input
    class="sc-hKwCoD hHLQXS"
    type="text"
    value=""
  />
</div>
`;
